ESP8266Interface.cpp

```diff
@@ -472,6 +480,34 @@ void ESP8266Interface::socket_attach(void *handle, void (*callback)(void *), voi
     _cbs[socket->id].data = data;
 }
 
+nsapi_error_t ESP8266Interface::setsockopt(nsapi_socket_t handle, int level,
+        int optname, const void *optval, unsigned optlen)
+{
+    struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
+    if (level == NSAPI_SOCKET) {
+        switch (optname) {
+            case NSAPI_KEEPALIVE: {
+                if(socket->connected == true) {// ESP8266 limitation, keepalive needs to be given before connecting
```

Comment:
Not a fan of saying `== true` - it's a tautology, and a potential error - `if (x == true)` is not always the opposite of `if (x == false)`. Best to stick with `if (x)` and `if (!x)` which are always opposites, and should be totally clear if your boolean variables are sensibly named - "if  socket connected" scans better than "if socket connected equals true".

Reply:
ok